### PR TITLE
Adds common exception class.

### PR DESCRIPTION
**Pull request by jana-rapava**

### CODE REVIEW

File: include/fty_common_db_exception.h [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+        LICENSING_ERR,
+        REQUEST_PARAM_BAD_ERR,
+        DATA_CONFLICT_ERR,
+    };
+
+    enum ErrorSubtype : uint8_t
+    {
+        DB_ERROR_UNKNOWN = 0,
+        BAD_INPUT_UNKNOWN = 0,
+        INTERNAL_UNKNOWN = 0,
```
Comment (jimklimov) [inline include/fty_common_db_exception.h]: are these `=0` intentional? how would code later differentiate these?

File: include/fty_common_db_exception.h [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+        virtual ~CommonException(){}
+
+        const char* what() const noexcept override;
+
+        inline int getStatus() const { return m_status; }
+        inline ErrorType getErrorType() const { return m_errorType; }
+        inline ErrorSubtype getErrorSubtype() const { return m_errorSubtype; }
+
+        //return an error payload from an exception
```
Comment (jimklimov) [inline include/fty_common_db_exception.h]: Small common nit, the spaces (and capital letters for first words in a sentence) after a `//` in many places in this PR

File: include/fty_common_db_exception.h (hunk 1 above)
Comment (jana-rapava) [reply include/fty_common_db_exception.h]: This class is intended to address this FIXME https://github.com/jana-rapava/fty-common-db/blob/master/include/fty_common_db_defs.h#L163 and provide an object-oriented exception-capable replacement for these enums (https://github.com/jana-rapava/fty-common-db/blob/master/include/fty_common_db_defs.h#L165 and down).

Zeroes are for backward compatibility.

File: include/fty_common_db_exception.h (hunk 2 above)
Comment (jana-rapava) [reply include/fty_common_db_exception.h]: I'm not sure if comments are (or should be) grammatically correct sentences with all the redundancies and unnecessary verbosity of natural languages :-)
Spaces definitely improve readability, I will add them.

**Comment by jimklimov** [comment]
Was all green before latest cosmetic change, so merging.

**Pull request merged**